ivy.el

```diff
@@ -3975,17 +3975,29 @@ characters (previous if ARG is negative)."
   "Overlay used to highlight yanked word.")
 
 (defvar ivy--pulse-timer nil
-  "Timer used to dispose of ivy--pulse-overlay.")
+  "Timer used to dispose of `ivy--pulse-overlay'.")
+
+(defvar ivy-pulse-delay 0.5
+  "Number of seconds to display `ivy-yanked-word' highlight.")
 
 (defun ivy--pulse-region (begin end)
   (if ivy--pulse-overlay
-      (move-overlay ivy--pulse-overlay begin end (current-buffer))
-    (setq ivy--pulse-overlay (make-overlay begin end)))
+      (move-overlay ivy--pulse-overlay
+                    (min begin (overlay-start ivy--pulse-overlay))
+                    (max end (overlay-end ivy--pulse-overlay)))
+    (setq ivy--pulse-overlay (make-overlay begin end))
+    (overlay-put ivy--pulse-overlay 'face 'ivy-yanked-word))
   (when ivy--pulse-timer
     (cancel-timer ivy--pulse-timer))
-  (move-overlay ivy--pulse-overlay begin end (current-buffer))
-  (overlay-put ivy--pulse-overlay 'face 'ivy-yanked-word)
-  (setq ivy--pulse-timer (run-at-time 0.5 nil #'delete-overlay ivy--pulse-overlay)))
+  (setq ivy--pulse-timer
+        (run-at-time ivy-pulse-delay nil #'ivy--pulse-cleanup)))
+
+(defun ivy--pulse-cleanup ()
+  "Cancel `ivy--pulse-timer' and delete `ivy--pulse-overlay'."
+  (when ivy--pulse-timer
+    (setq ivy--pulse-timer (cancel-timer ivy--pulse-timer)))
+  (when ivy--pulse-overlay
+    (setq ivy--pulse-overlay (delete-overlay ivy--pulse-overlay))))
```

Comment:
Does delete overlay return anything meaningful?

Its docstring does not make that clear.

Why not set that to nil?

Reply:
> Does delete overlay return anything meaningful?

It always returns `nil`. It's not documented as doing that, but it's unlikely to change, so I just used it to save myself having to write a second line. Same with `cancel-timer`. Feel free to change that if it irks you.

Reply:
You are free to work on anything you like and suggest any changes you like. Same goes for any other contributor around here. There is nothing unhealthy about that; it's what many people find most fun about FOSS. Each person has different itches.

Re: the nitpicking - either you have misunderstood my intention, or my tastes have changed. I'm only human, you know. I don't recall ever saying anything about a guaranteed return value in a context where the intention is clear, but it's definitely possible. I'm sorry you find it so frustrating to work on the same project as me.

Reply:
@mookid I forgot to thank you for your efforts in fixing Ivy bugs, it's much appreciated.